docs/runbook/formulare-support.md

```diff
@@ -136,7 +136,37 @@ Treba kontaktovať vlastníkov formuláru na strane magistrátu. Ak ide o závä
+
+Postup pri zlom roku je nasledovný:
+1. Otvorím si daný formulár v databáze, a vo `formDataJson` upravím rok tak, aby nebol pod 1900. Vo väčšine prípadov stačí zmeniť 18xx -> 19xx, prípadne niekto zabudne dať prvú cifru, teda 9xx -> 19xx.
+2. Rovnako tomuto formuláru nastavím `ginisState: SUBMISSION_ASSIGNED`.
+3. Otvorím [sharepoint](https://magistratba.sharepoint.com/sites/UsmernovanieInvesticnejCinnosti_prod/_layouts/15/viewlsts.aspx?view=14) a prejdem si všetky tabuľky, z ktorých treba vymazať všetky záznamy pre danú žiadosť ktoré sa už do sharepointu dostali. Do sharepointu posielame veci postupne, teda sa môže stať že nejaké záznamy pre túto žiadosť už sú v sharepointe. Stačí v tabuľke vyhľadať záznamy podľa Ginis ID, ak sa tam nejaký nachádza tak ho treba odstrániť, ak nie tak netreba robiť nič.
+4. Ak je všetko zo sharepointu odstránené a formulár je fixnutý aj so správnym `ginisState`, tak ho stačí hodiť do Ginis queue ako je popísané vyššie. Následne prebehne pokus o odoslanie, ak je všetko v poriadku tak sa dostane do stavu `PROCESSING`.
+
+Ak toto nebol ten problém a roky sú teda všetky v poriadku, tak treba skúšať ďalej. Niekedy sa môže stať, že Sharepoint timeoutne, toto sa stalo asi iba raz, ale v tom prípade treba spraviť rovnaké kroky ako vyššie, len netreba upravovať dáta.
+
+Párkrát sa stalo, že nesedel dátový typ ktorý bol odoslaný s tým, ktorý bol v sharepointe pre daný stĺpec. Všetky problémy s týmto boli, že bol nastavený príliš malý limit na dĺžku textov. V tom prípade treba zmeniť stĺpec v sharepointe z "jedného riadku textu" na "viac riadkov textu" a znova zopakovať odoslanie.
+
+Ak stále nič nefunguje, tak sa dá zreprodukovať odosielanie aj lokálne, sharepoint vráti nejaký error log, ktorý je nie vždy veľa hovoriaci, ale vie niekedy pomôcť. Na toto treba mať nejaký tool na posielanie requestov, napríklad postman.
+1. Na toto treba získať auth token. Treba si nastaviť env hodnoty `SHAREPOINT_TENANT_ID, SHAREPOINT_CLIENT_ID, SHAREPOINT_CLIENT_SECRET, SHAREPOINT_DOMAIN, SHAREPOINT_URL` na produkčné hodnoty a nejak lokálne zavolať `getAccessToken` v `sharepoint.subservice`. Funkcia vráti access token, s ktorým vieme simulovať potom posielanie do sharepointu.
+2. V logoch je vidno, že do akej tabuľky/zoznamu zlyhalo odoslanie, má to tvar `dtb_NajomneByvanie...` a aj aké dáta sa tam poslali.
+3. Ak je napríklad tabuľka `dtb_NajomneByvanieZiadatel`, tak budeme posielať POST request na `https://magistratba.sharepoint.com/sites/UsmernovanieInvesticnejCinnosti_prod/_api/web/lists/getbytitle('dtb_NajomneByvanieZiadatel')/items`. Pre iný zoznam stačí samozrejme zmeniť za iný v url v `getbytitle`.
+4. Autorizujeme sa bearer tokenom ktorý máme z prvého kroku.
```

Comment:
You call it auth token in the first step, and bearer token in the fourth. Unless I misunderstand some nuance here, it could be unified.